triggers/scheduler.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import typer
+from beanie import init_beanie
+from motor.motor_asyncio import AsyncIOMotorClient
+from rocketry import Rocketry
+from rocketry.conds import daily
+
+from api.adapters.utils import RepositoryType, get_adapter_by_type
+from api.config import get_settings
+from api.models.catalog import DatasetMetadataDOC
+from api.models.user import Submission
+
+app = Rocketry(config={"task_execution": "async"})
+logger = logging.getLogger()
+
+
+async def retrieve_repository_record(submission: Submission):
+    """Retrieve the repository record from the repository and return it as a Catalog Record"""
+    if submission.repository in RepositoryType.__members__:
+        adapter = get_adapter_by_type(RepositoryType[submission.repository])
+        try:
+            repo_record_metadata = await adapter.get_metadata(submission.repository_identifier)
+        except Exception as err:
+            if hasattr(err, "detail"):
+                err_details = err.detail
+            else:
+                err_details = str(err)
+
+            err_msg = (f"Error retrieving metadata from {submission.repository} repository "
+                       f"for record id: {submission.repository_identifier}: {err_details}")
+            logger.error(err_msg)
+            return None
+        return adapter.to_catalog_record(repo_record_metadata)
+    else:
+        err_msg = f"Repository type {submission.repository} is not supported"
+        logger.error(err_msg)
+        raise Exception(err_msg)
+
+
+@app.task(daily)
+async def do_daily():
+    settings = get_settings()
+    db = AsyncIOMotorClient(settings.db_connection_string)[settings.database_name]
+    await init_beanie(database=db, document_models=[Submission, DatasetMetadataDOC])
+
+    async for submission in Submission.find(Submission.repository != None):
+        try:
+            dataset = await DatasetMetadataDOC.get(submission.identifier)
+            if dataset is None:
+                # TODO: delete submission record?
+                # await submission.delete()
+                # logger.warning(f"Deleting Submission {submission.url} as it is not in catalog")
```

Comment:
This is a case that should not happen and indicates that there is an issue with the data integrity.  I do not think we should delete the submission.  I do want to be aware if this case happens.

I also don't know if we want to delete the discovery record because if we do get to this point and the discovery record exists, we can repair the submission by copying the discovery record to the catalog.

Long story short, I think we should log a warning then move on without making any changes to the collections.